contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,332 @@
+     * - user must not be borrower
+     * @param _borrower the address that will interact with contract
+     */
+    function allowBorrower(address _borrower) external onlyGovernor {
+        if (borrowers[_borrower].isAllowed) revert IncurDebtV1_AlreadyBorrower(_borrower);
+        borrowers[_borrower].isAllowed = true;
+    }
+
+    /**
+     * @notice sets the maximum debt limit for a borrower
+     * - onlyOwner (or governance)
+     * - limit must be greater than or equal to borrower's outstanding debt
+     * - limit must be less than or equal to the global debt limit
+     * @param _borrower the address that will interact with contract
+     * @param _limit borrower's debt limit in OHM
+     */
+    function setBorrowerDebtLimit(address _borrower, uint256 _limit) external onlyGovernor isBorrower(_borrower) {
+        if (_limit < borrowers[_borrower].debt) revert IncurDebtV1_AboveBorrowersDebtLimit(_limit);
+        if (_limit > globalDebtLimit) revert IncurDebtV1_AboveGlobalDebtLimit(_limit);
+
+        borrowers[_borrower].limit = uint128(_limit);
+    }
+
+    /**
+     * @notice revoke user right to borrow
+     * - onlyOwner (or governance)
+     * - user must be borrower
+     * - borrower must not have outstanding debt
+     * @param _borrower the address that will interact with contract
+     */
+    function revokeBorrower(address _borrower) external onlyGovernor isBorrower(_borrower) {
+        if (borrowers[_borrower].debt != 0) revert IncurDebtV1_BorrowerStillHasOutstandingDebet(_borrower);
+        borrowers[_borrower].isAllowed = false;
+    }
+
+    /**
+     * @notice deposits gOHM/sOHM to use as collateral
+     * - msg.sender must be a borrower
+     * - this contract must have been approved _amount
+     * @dev will unwrap and hold as sOHM in this contract
+     * @param _amount amount of gOHM/sOHM
+     * @param _token token(gOHM/sOHM) to deposit with
+     */
+    function deposit(uint256 _amount, address _token) external isBorrower(msg.sender) {
+        Borrower storage borrower = borrowers[msg.sender];
+        IERC20(_token).safeTransferFrom(msg.sender, address(this), _amount);
+
+        if (_token == gOHM) {
+            uint256 sbalance = IStaking(staking).unwrap(address(this), _amount);
+
+            borrower.collateralInSOHM += uint128(sbalance);
+            borrower.collateralInGOHM += _amount;
+        } else if (_token == sOHM) {
+            uint256 gbalance = IgOHM(gOHM).balanceTo(_amount);
+
+            borrower.collateralInSOHM += uint128(_amount);
+            borrower.collateralInGOHM += gbalance;
+        }
+    }
+
+    /**
+     * @notice allow borrowers to borrow OHM
+     * - msg.sender must be a borrower
+     * - _ohmAmount must be less than or equal to borrowers debt limit
+     * - _ohmAmount must be less than or equal to borrowers available loan limit
+     * @param _ohmAmount amount of OHM to borrow
+     */
+    function borrow(uint256 _ohmAmount) external isBorrower(msg.sender) {
```

Comment:
is this implementation the tokemake version where borrower just gets ohm? If so we should implement that interface.

Reply:
about the interfaces i made comment on the other pr
https://github.com/OlympusDAO/olympus-contracts/pull/252

Suggesting we merge the interfaces and make one contract that implements both since they quite similar and i think it might be very low friction having both functionalities in 1 contract. Just some permissioning.

Reply:
That would mean this function will need another permission aside from isAllowed() if we go down that route.

Reply:
> about the interfaces i made comment on the other pr #252
> 
> Suggesting we merge the interfaces and make one contract that implements both since they quite similar and i think it might be very low friction having both functionalities in 1 contract. Just some permissioning.

Perhaps, that's up to the policy to decide tho.

Reply:
> That would mean this function will need another permission aside from isAllowed() if we go down that route.

Yeah

Reply:
why does policy decide this i though this would be an engineering problem. As long as functionality same they won't care right?

Reply:
Yes agreed with paul. just make a common interface for all cases. no use splitting it imo.